tests/test_csc.py

```diff
@@ -0,0 +1,302 @@
+            self.patch.return_value.AVS_Activate.assert_called_once()
+
+            # In DISABLED, there should still be a connected device.
+            await harness.remote.cmd_disable.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.DISABLED)
+            self.assertIsNotNone(harness.csc.device)
+            # we only should try to activate once in this whole sequence
+            self.patch.return_value.AVS_Activate.assert_called_once()
+
+            # in STANDBY, there should be no connected device
+            await harness.remote.cmd_standby.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.STANDBY)
+            self.assertIsNone(harness.csc.device)
+            # Entering standby should close the connection; Done might be
+            # called twice: once for the explicit `disconnect()` and  once
+            # for the destruction of the device object (but due to python
+            # garbage collection, the latter is not guaranteed to have happened
+            # immediately).
+            self.patch.return_value.AVS_Done.assert_called()
+
+    async def test_enable_fails(self):
+        """Test that exceptions raised when connecting cause a fault when
+        switching the CSC from STANDBY to DISABLED.
+        """
+        self.patch.return_value.AVS_Activate.return_value = AvsReturnCode.invalidHandle.value
+        async with Harness(initial_state=salobj.State.STANDBY) as harness:
+            # Check that we are properly in STANDBY at the start
+            await self.check_summaryState(harness.remote, salobj.State.STANDBY)
+
+            msg = "Failed to connect"
+            with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED, result_contains=msg):
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+            state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(state.summaryState, salobj.State.FAULT)
+            error = await harness.remote.evt_errorCode.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertIn("RuntimeError('Invalid device handle; cannot activate device", error.errorReport)
+            self.assertIsNone(harness.csc.device)
+
+    async def test_expose(self):
+        """Test that we can take an exposure and that appropriate events are
+        emitted.
+        """
+        async with Harness(initial_state=salobj.State.ENABLED) as harness:
+            # Check that we are properly in ENABLED at the start
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+
+            duration = 2  # seconds
+            task = asyncio.create_task(harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT+duration,
+                                                                           duration=duration))
+            await asyncio.sleep(0.1)  # give the event loop time to start
```

Comment:
Why wait? `check_exposureState` should still work because it waits for the event.